river/InputConfig.zig

```diff
@@ -137,6 +137,23 @@ pub const DwtState = enum {
     }
 };
 
+pub const DwtpState = enum {
+    disabled,
+    enabled,
+
+    pub fn apply(dwtp_state: DwtpState, device: *c.libinput_device) void {
+        const want = @as(c_uint, switch (dwtp_state) {
+            .disabled => c.LIBINPUT_CONFIG_DWTP_DISABLED,
+            .enabled => c.LIBINPUT_CONFIG_DWTP_ENABLED,
+        });
+        if (c.libinput_device_config_dwtp_is_available(device) == 0) return;
+        const current = c.libinput_device_config_dwtp_get_enabled(device);
+        if (want != current) {
+            _ = c.libinput_device_config_dwtp_set_enabled(device, want);
+        }
```

Comment:
From skimming the libinput source code, it doesn't seem necessary to call any libinput functions other than `libinput_device_config_dwtp_set_enabled()` here. If it's not supported or there is no state change libinput will do nothing.

Reply:
Hmm, looks like we do a lot of redundant libinput getter calls in this file...

Reply:
IIRC there was a reason why I checked the current state of all libinput settings before changing them, but i can't remember it.